include/d/d_stage.h

```diff
@@ -0,0 +1,177 @@
+#ifndef D_STAGE_H
+#define D_STAGE_H
+
+#include "common.h"
+#include "d/d_base.h"
+#include "d/d_fader.h"
+#include "d/d_room.h"
+#include "m/m_allocator.h"
+#include "m/m_mtx.h"
+#include "m/m_vec.h"
+#include "s/s_FPhase.h"
+
+#define MAX_ROOM_NUMBER 64
+
+class dRoomTable_c {
+public:
+    /* 801b4670 */ dRoomTable_c();
+    virtual ~dRoomTable_c() {}
+
+    void setRoom(int roomid, dRoom_c *room) {
+        mRooms[roomid] = room;
+    }
+
+    dRoom_c *getRoom(s32 idx) {
+        return mRooms[idx];
+    }
+
+private:
+    dRoom_c *mRooms[MAX_ROOM_NUMBER - 1];
+};
+
+class EggScreenEffectBlur {
+public:
+    EggScreenEffectBlur();
+    ~EggScreenEffectBlur();
+
+private:
+    u8 _0x00[0x38 - 0x00];
+};
+
+class EggPostEffectBlur {
+public:
+    EggPostEffectBlur();
+    ~EggPostEffectBlur() {}
+
+    void setField0x30(f32 val) {
+        field_0x30 = val;
+    }
+
+private:
+    u8 _0x00[0x30 - 0x00];
+    f32 field_0x30;
+    u8 _0x03[0x78 - 0x34];
+};
```

Comment:
Im not sure if I went back and fixed these types. When i was noting ghidra at one point, I noticed that the Post Effects and Screen Effects I did not have enough info for. 

Blur is probably correct tho lol

Reply:
Moved these types to their Egg headers for now